scarlet/operator.py

```diff
@@ -259,24 +309,55 @@ def prox_uncentered_symmetry(X, step, center=None, strength=.5, use_soft=True, f
         The parameter to update.
     step: `int`
         Step size of the gradient step.
-    center: tuple
+    center: tuple of `int`
         The center pixel coordinates to apply the symmetry operator.
+    algorithm: `string`
+        The algorithm to use for symmetry.
+        * If `algorithm = "kspace" then `X` is shifted by `shift` and
+          symmetry is performed in kspace. This is the only ymmetry algorithm
+          in scarlet that works for fractional pixel shifts.
+        * If `algorithm = "sdss" then the SDSS symmetry is used,
+          namely the source is made symmetric around the `center` pixel
+          by taking the minimum of each pixel and its symmetric partner.
+          This is the algorithm used when initializing an `ExtendedSource`
+          because it keeps the morphologies small, but during optimization
+          the penalty is much stronger than the gradient
+          and often leads to vanishing sources.
+        * If `algorithm = "soft" then soft symmetry is used,
+          meaning `X` will be allowed to differ from symmetry by the fraction
+          `strength` from a perfectly symmetric solution. It is advised against
+          using this algorithm because it does not work in general for sources
+          shifted by a fractional amount, however it is used internally if
+          a source is centered perfectly on a pixel.
+    fill: `float`
+        The value to fill the region that cannot be made symmetric.
+        When `fill` is `None` then the region of `X` that is not symmetric
+        is not constrained.
     strength: `float`
         The amount that symmetry is enforced. If `strength=0` then no
         symmetry is enforced, while `strength=1` enforces strict symmetry
         (ie. the mean of the two symmetric pixels is used for both of them).
-    use_soft: `bool`
-        Whether to use the strict SDSS symmetry `use_soft=False` or
-        the soft proximal operator (`use_soft=True`).
+        This parameter is only used when `algorithm = "soft"`.
 
     Returns
     -------
     result: `function`
         The update function based on the specified parameters.
     """
-    if use_soft:
+    if algorithm == "kspace" and (shift is None or np.all(shift == 0)):
+        algorithm = "soft"
+        strength = 1
+    if algorithm == "kspace":
+        return uncentered_operator(X, prox_kspace_symmetry, center, shift=shift, step=step, fill=fill)
+    if algorithm == "sdss":
```

Comment:
`elif` ?

Reply:
I *think* that it is more canonical to use another `if` when the previous `if` statement will always return from the function.

Reply:
Right, I missed that